File: frontend/src/types/logs.ts
export type LogLevel = 'DEBUG' | 'INFO' | 'WARNING' | 'ERROR' | 'CRITICAL';

export interface LogEntry {
  timestamp: string;
  level: LogLevel;
  logger: string;
  message: string;
  exc_info?: string;
}

export interface LogsResponse {
  entries: LogEntry[];
  total: number;
}

export interface LogsQuery {
  level?: LogLevel | '';
  keyword?: string;
  limit?: number;
}

File: src/kint/schemas/logs.py
"""ログAPIスキーマ定義。"""

from pydantic import BaseModel


class LogEntry(BaseModel):
    """1件のログエントリ。"""

    timestamp: str
    level: str
    logger: str
    message: str
    exc_info: str | None = None


class LogsResponse(BaseModel):
    """ログ一覧レスポンス。"""

    entries: list[LogEntry]
    total: int
